projects/storefrontlib/src/lib/cms/services/cms-routes.service.ts

```diff
@@ -0,0 +1,68 @@
+import { Injectable } from '@angular/core';
+import { Route, Router } from '@angular/router';
+import { CmsRoute, CmsService, PageContext, PageType } from '@spartacus/core';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { PageLayoutComponent } from '../page-layout/page-layout.component';
+import { CmsMappingService } from './cms-mapping.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CmsRoutesService {
+  constructor(
+    private router: Router,
+    private cmsService: CmsService,
+    private cmsMapping: CmsMappingService
+  ) {}
+
+  contentRouteExist(url: string): boolean {
+    return !!this.router.config.find(
+      (route: CmsRoute) =>
+        route.data &&
+        route.data.cxCmsContext &&
+        url.startsWith('/') &&
+        route.path === url.substr(1)
+    );
+  }
+
+  handleContentRoutes(
+    pageContext: PageContext,
+    currentUrl: string
+  ): Observable<boolean> {
+    return this.cmsService.getPageState(pageContext).pipe(
+      map(pageData => {
+        const componentRoutes = this.cmsMapping.getRoutesFromPageData(pageData);
+        if (componentRoutes.length) {
+          if (this.updateRouting(pageContext, componentRoutes)) {
+            this.router.navigateByUrl(currentUrl);
+            return false;
+          }
+        }
+        return true;
+      })
+    );
+  }
+
+  private updateRouting(pageContext: PageContext, routes: Route[]): boolean {
+    if (
+      pageContext.type === PageType.CONTENT_PAGE &&
```

Comment:
What about Product/Category/(Catalog) pages? Will be handled in separate ticket?

Reply:
Yep, as we discussed with Tobias, it is not part of this ticket. We should also probably first decide how we want to get current producrCode/categoryCode/etc. in our app.